buildSrc/src/main/kotlin/pklAllProjects.gradle.kts

```diff
@@ -1,10 +1,50 @@
+@file:Suppress("UnstableApiUsage")
+
 import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
 
 val buildInfo = extensions.create<BuildInfo>("buildInfo", project)
 
-dependencyLocking {
-  lockAllConfigurations()
-}
+// Default JVM bytecode target, and Java language level.
+private val defaultJavaTarget = "11"
+
+// Property where the JVM target and Java language level can be overridden.
+private val buildPropertyJavaTarget = "javaTarget"
+
+// Property where the Kotlin target can be overridden.
+private val buildPropertyKotlinTarget = "kotlinTarget"
+
+// Property to enable dependency locking.
+private val buildPropertyDependencyLocks = "lockDependencies"
+
+// Property to enable strict compiler modes.
+private val buildPropertyStrictMode = "strict"
+
+// Arguments to pass to `javac`.
+private val javacArgs = listOf(
+  "-parameters",
+)
+
+// Arguments to pass to `kotlinc`.
+private val kotlincArgs = listOf(
+  "-Xjsr305=strict",
+  "-Xjvm-default=all",
+)
+
+// Configurations which are subject to dependency locking.
+private val lockedConfigurations = listOf(
+  "annotationProcessor",
+  "archives",
+  "compileClasspath",
+  "graal",
+  "kotlinCompilerClasspath",
+  "kotlinKlibCommonizerClasspath",
+  "runtimeClasspath",
+  "testCompileClasspath",
+  "testRuntimeClasspath",
+  "truffle",
+)
```

Comment:
This is the list of configurations which have locked dependencies now. Should prevent Spotless and other detached configurations from polluting the Gradle locks.

Reply:
@stackoverflow Would the team be open to a smaller PR which applies this change above ☝🏻 to fix the lockfile issue surfacing in #194?

Reply:
Smaller PRs would definitely help smoother review+merge processes. This is _not at all_ to discourage you from the _lovely_ work you're doing!